custom_components/smartthinq_sensors/wideq/devices/ac.py

```diff
@@ -1247,6 +1251,39 @@ def filters_life(self):
 
         return result
 
+    @property
+    def pm1(self):
+        """Return Air PM1 value."""
+        support_key = self._get_state_key(SUPPORT_AIR_POLUTION)
+        if self._device.model_info.enum_value(support_key, "@PM1_0_SUPPORT") is None:
+            return None
```

Comment:
I suggest here to create a new function at device level, eg. `is_pm_supported` (similar, for example to `is_mode_airclean_supported`), to avoid to scan every status refresh `model_info`.

You could start creating a new constant:
```
SUPPORT_PM = [SUPPORT_RAC_MODE, ["@PM1_0_SUPPORT", "@PM10_SUPPORT", "@PM2_5_SUPPORT"]}
```

In the new function you should loop the second array and createa new boolean array  (and store this in a new device attribute) that represent supported PM type.

Then here you can just call the device method that will scan `model_info` only on the first call.

Reply:
As commited in 7cb81de.

Do you have any suggestion how to debug this component outside homeassistant?

Reply:
- Install [Docker Desktop](https://docs.docker.com/desktop/) and [Visual studio code](https://code.visualstudio.com/)
- Open the project with Visual studio code
- When the project is open, select the option `clone in a new volume`
- A new environment will be created, and you can run a copy of HA with integration for test.

Do you want that I wait for your test result before merging?